### PR TITLE
implement iron-validatable-behavior

### PR DESCRIPTION
**Pull request by morethanreal**
@cdata @notwaldorf PTAL

### CODE REVIEW

File: demo/cats-only.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+-->
+
+<link rel="import" href="../../polymer/polymer.html">
+<link rel="import" href="../../iron-validator-behavior/iron-validator-behavior.html">
+
+<script>
+
+  Polymer({
+
+    is: 'cats-only',
```
Comment (notwaldorf) [inline demo/cats-only.html:18]: :smile_cat:

**Comment by notwaldorf** [comment]
So this is a behaviour that I would put on top of an `iron-input`? Could the `invalid` state surface to the `paper-input` later as well?

**Comment by morethanreal** [comment]
I'm going to bake that into `iron-input`. Yeah, definitely thinking about whether `invalid` should be part of the API.

**Comment by morethanreal** [comment]
Added `invalid`. PTAnotherL?

**Comment by notwaldorf** [comment]
This is really neat! LGTM

**Pull request merged**